### PR TITLE
Use monai instead of monai-weekly in the requirements

### LINKED ISSUE #386: Fixed MONAI version in requirements may disrupt developer's environment
In the requirement, the version of MONAI is fixed to:
```
monai-weekly==1.2.dev2304
```

Maybe we can provide a bit more flexibility for the developers by:

```
monai-weekly>=1.2.dev2304
```

### PR DESCRIPTION
**Pull request by mingxin-zheng**
Fix #386

### CODE REVIEW

**Comment by ericspod** [comment]
We do need to change the requirements mentioned in `setup.py` as well. With that changed this should work, though I'd suggest we reorganise things in the future to use `pyproject.toml` exclusively to define dependencies. I point people to [RandomDataset](https://github.com/KCL-BMEIS/RandomDataset) as an example of setting up a repo like this which I think is the more modern way.

**Comment by Warvito** [comment]
As @ericspod mentioned, it is necessary to change the following line to update the deps

https://github.com/Project-MONAI/GenerativeModels/blob/01a46549669b1e97da66db822d59a66cc105ddd8/setup.py#L21

I also agree with Eric regarding the future reorganisation of setup.py and pyproject.toml to be more similar to the other monai packages and randomDataset. I will create an issue for that

**Comment by ericspod** [comment]
We might want to bump version to 0.2.2 with these fixes as well so that the changes in tutorials can reference that fixed version.

**Comment by mingxin-zheng** [comment]
Thank you @ericspod @Warvito for the comments. Just updated the PR accordingly.

File: setup.py [hunk 1]
```diff
@@ -18,5 +18,5 @@
     packages=find_packages(exclude=[]),
     version="0.2.1",
```
Comment (ericspod) [inline setup.py:19]:
```suggestion
    version="0.2.2",
```
Let's bump the version and do a release to be sure it's working correctly.

**Review by Warvito — APPROVED** [review]

**Pull request merged**